include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -266,6 +315,9 @@ class counting_iterator
         return !(*this < __it);
     }
 
+    friend std::true_type
+    is_onedpl_indirectly_device_accessible_iterator(const counting_iterator&);
```

Comment:
Does it make sense to define this function here as `constexpr` ?
I mean really their result doesn't depends from `counting_iterator` state...

Reply:
I originally had this as `constexpr` and removed it.  We deliberately don't define a body for the function because it is only used in a `decltype` context to determine the return type (which must be known / available at compile-time inherently).  

It may seem to be more descriptive with `constexpr`, but tend to I think it may be somewhat misleading.  This function should never be called outside a `decltype` context, and `constexpr` doesn't **require** that a function is only used at compile-time, rather it says that this is available to be called at compile-time.  In fact we aren't ever calling the function even at compile-time,  we are merely deducing the return type, which you can do with any function whether it is labelled  `constexpr` or not.